contracts/minter/src/contract.rs

```diff
@@ -377,27 +399,48 @@ fn _execute_mint(
     };
 
     // calculate the mint fee
+    // if in whitelist, use whitelist price. else use mint price
+    let mint_price: Coin = if let Some(whitelist) = config.whitelist {
+        let res_started: HasStartedResponse = deps
+            .querier
+            .query_wasm_smart(whitelist.clone(), &WhitelistQueryMsg::HasStarted {})?;
+        let res_ended: HasEndedResponse = deps
+            .querier
+            .query_wasm_smart(whitelist.clone(), &WhitelistQueryMsg::HasEnded {})?;
+        if res_started.has_started && !res_ended.has_ended {
+            let unit_price: UnitPriceResponse = deps
+                .querier
+                .query_wasm_smart(whitelist, &WhitelistQueryMsg::UnitPrice {})?;
+            unit_price.unit_price
+        } else {
+            config.unit_price.clone()
+        }
+    } else {
+        config.unit_price.clone()
+    };
```

Comment:
can we refactor into a method  seems like its the same code for batch and regular mint.

Reply:
yes i was thinking the same!